### PR TITLE
*: refactor iterator interface

### PR DESCRIPTION
**Pull request by c4pt0r**
Next() shouldn't return a new iterator.
All tests passed.

### CODE REVIEW

**Comment by c4pt0r** [comment]
@ngaut @siddontang PTAL

File: kv/btree_buffer.go [hunk 1]
```diff
@@ -93,10 +93,10 @@ func (i *btreeIter) Value() []byte {
 }
 
 // Next implements Iterator Next.
-func (i *btreeIter) Next() (Iterator, error) {
+func (i *btreeIter) Next() error {
 	k, v, err := i.e.Next()
 	i.k, i.v, i.ok = string(fromIfaces(k)), fromIfaces(v), err == nil
-	return i, err
+	return err
```
Comment (siddontang) [inline kv/btree_buffer.go]: errors.Trace(err)

**Comment by ngaut** [comment]
LGTM

File: kv/btree_buffer.go [hunk 2]
```diff
@@ -93,10 +99,10 @@ func (i *btreeIter) Value() []byte {
 }
 
 // Next implements Iterator Next.
-func (i *btreeIter) Next() (Iterator, error) {
+func (i *btreeIter) Next() error {
 	k, v, err := i.e.Next()
 	i.k, i.v, i.ok = string(fromIfaces(k)), fromIfaces(v), err == nil
-	return i, err
+	return errors.Trace(err)
```
Comment (qiuyesuifeng) [inline kv/btree_buffer.go]: I prefer to check error in line 103

Comment (c4pt0r) [inline kv/btree_buffer.go]: DONE

**Comment by c4pt0r** [comment]
@siddontang  PTAL

**Comment by siddontang** [comment]
LGTM

**Pull request merged**